IotHubConsumer/Receiver.cs

```diff
@@ -45,7 +45,12 @@ await foreach (var partitionEvent in consumer.ReadEventsAsync(cancellationToken)
                         retainFlag = partitionEvent.Data.Properties["mqtt-retain"].ToString();
                     }
 
-                    receivedMessages.Add(new D2CMessage { Payload = data, RetainFlag = retainFlag });
+                    var messageType = "telemetry";
+                    if (partitionEvent.Data.Properties.ContainsKey("iothub-MessageType"))
+                    {
+                        messageType = partitionEvent.Data.Properties["iothub-MessageType"].ToString();
+                    }
+                    receivedMessages.Add(new D2CMessage { Payload = data, RetainFlag = retainFlag, MessageType = messageType });
```

Comment:
Now that you've added a second type, do you still think we should be using a object as opposed to sending all the properties?

Reply:
@Ioana37 Is this type `D2CMessage` a replacement of the previous type? In that case I'm fine with only having the specific properties. If we send all properties, then we'll have to parse them in the lambda expression inside the unit test.

Reply:
@syedhassaanahmed yes, I renamed it to make it more readable. I actually thought to use the whole object, but it has a pretty complex structure - what we need are subproperties of another property, and what is more, if they are not set to true, they do not appear at all.